### ISSUE TITLE
Feature request: Support XML output

### ISSUE THREAD

**Issue by pdumais** [freeconf/restconf#33]
I'm not 100% sure if this is RFC compliant but I think the logic would be valid:
Restconf could default to always return data with yang-data+json.
Unless the client sends "Accept: yang-data+xml" and nothing else. In this case, restconf would send the reply in xml.

bascically:
| Client Accept Header | What restconf returns |
|--- | --- |
|yang-data+json | yang-data+json|
|yang-data+json and yang-data+xml | yang-data+json|
|yang-data+xml | yang-data+xml|
|something else | 415 ? |

So the logic is only based on the Accept header and not on the Content-Type.

**Comment by dhubler** [comment]
Simply put
   If FreeCONF is asked to Accept or Respond in XML then I think FreeCONF
should send error.
Until
  FreeCONF supports XML

**Comment by pdumais** [comment]
technically yes, but it would be better for us to continue to silently process it.
If I can have it my way, I would keep status quo until XML is supported. And I would say #32 would be more useful to start with.

**Comment by dhubler** [comment]
#32 assumes you can read XML and today you cannot.  Your assumption
FreeCONF can currently read XML I think is incorrect.

**Comment by pdumais** [comment]
No, the other issue I raised had a comment abput that but it was incorrect. #32 is about adding functionality to parse xml.

I created 2 requests: one to read, the other to write.

**Comment by dhubler** [comment]
when we had read and write
   solution is obvious
when we have one w/o the other
  i assume this is temporary and if you really need the branch on freeconf
github, we can keep that on a branch and you can do whatever you'd like

**Comment by pdumais** [comment]
I just thought it was easier to divide the work in two. Since one item is about checking the contentType and acting on it. The other is about checking the Accept header.

If you prefer 1 single task, I'm good with it too It's your project :) 
But I'm gonna explore how to parse XML on my side in the meantime.

**Comment by dhubler** [comment]
dividing the work is always good

our project.  my philosophy is once someone contributes to a certain level,
they get to do whatever they want.  you could still convince me 1/2 is
better than none in master, but i suspect once you do half, you'll want to
finish and this won't be an issue.
